presto-main/src/main/java/com/facebook/presto/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -287,26 +302,43 @@ public static Block regexpSplit(@SqlType("varchar(x)") Slice source, @SqlType(Jo
 
         int lastEnd = 0;
         int nextStart = 0;
-        while (true) {
-            int offset = matcher.search(nextStart, source.length(), Option.DEFAULT);
-            if (offset == -1) {
-                break;
-            }
+        int offset = getMatchingOffset(matcher, nextStart, source.length());
+        if (offset == -1) {
+            VARCHAR.writeSlice(blockBuilder, source);
+            return blockBuilder.build();
+        }
+
+        do {
             if (matcher.getEnd() == matcher.getBegin()) {
                 nextStart = matcher.getEnd() + 1;
             }
             else {
                 nextStart = matcher.getEnd();
             }
-            Slice slice = source.slice(lastEnd, matcher.getBegin() - lastEnd);
+            VARCHAR.writeSlice(blockBuilder, source, lastEnd, matcher.getBegin() - lastEnd);
             lastEnd = matcher.getEnd();
-            VARCHAR.writeSlice(blockBuilder, slice);
-        }
-        VARCHAR.writeSlice(blockBuilder, source.slice(lastEnd, source.length() - lastEnd));
+            offset = getMatchingOffset(matcher, nextStart, source.length());
+        } while (offset != -1);
 
+        VARCHAR.writeSlice(blockBuilder, source.slice(lastEnd, source.length() - lastEnd));
         return blockBuilder.build();
     }
 
+    private static int getMatchingOffset(Matcher matcher, int at, int range)
+    {
+        return getMatchingOffset(matcher, at, range, true);
+    }
+
+    private static int getMatchingOffset(Matcher matcher, int at, int range, boolean noGroups)
+    {
+        try {
+            return matcher.searchInterruptible(at, range, noGroups ? DONT_CAPTURE_GROUP : DEFAULT);
+        }
+        catch (InterruptedException interruptedException) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)